audbackend/core/artifactory.py

```diff
@@ -1,13 +1,143 @@
 import os
+import typing
 
+import artifactory
 import dohq_artifactory
 
-import audfactory
+import audeer
 
 from audbackend.core import utils
 from audbackend.core.backend import Backend
 
 
+def _artifactory_path(
+        path,
+        username,
+        apikey,
+) -> artifactory.ArtifactoryPath:
+    r"""Authenticate at Artifactory and get path object."""
+    return artifactory.ArtifactoryPath(
+        path,
+        auth=(username, apikey),
+    )
+
+
+def _authentication(host) -> typing.Tuple[str, str]:
+    """Look for username and API key.
+
+    Looks for the two environment variables
+    ``ARTIFACTORY_USERNAME`` and
+    ``ARTIFACTORY_API_KEY``.
+
+    Missing values are extracted them from the
+    :file:`~/.artifactory_python.cfg` config file.
+
+    If it cannot find the config file
+    or a matching entry in the config file
+    it will set the username to ``anonymous``
+    and the API key to an empty string.
+    If your Artifactory server is configured
+    to allow anonymous users
+    you will be able to access the server this way.
+
+    Args:
+        host: host address
+
+    Returns:
+        username and API key
+
+    """
+    username = os.getenv('ARTIFACTORY_USERNAME', None)
+    apikey = os.getenv('ARTIFACTORY_API_KEY', None)
+
+    if apikey is None or username is None:  # pragma: no cover
+
+        config_entry = artifactory.get_global_config_entry(host)
```

Comment:
I checked and the function already takes care of removing `http://` or `https://`.